File: Data Structures & Algorithms/spiral-matrix/submission-1.cs
public class Solution {
    public List<int> SpiralOrder(int[][] matrix) {
        var ret = new List<int>();
        int top = 0, bottom = matrix.Length;
        int left = 0, right = matrix[0].Length;

        while(ret.Count < matrix.Length * matrix[0].Length)
        {
            for (int i = left ; i < right ; i++)
            {
                ret.Add(matrix[top][i]);
            }top++;

            for (int i = top ; i < bottom ; i++)
            {
                ret.Add(matrix[i][right - 1]);
            }right--;

            if (left >= right || top >= bottom)   return ret;

            for (int i = right  - 1; i >= left ; i--)
            {
                ret.Add(matrix[bottom - 1][i]);
            }bottom--; 

            for (int i = bottom - 1; i >= top ; i--)
            {
                ret.Add(matrix[i][left]);
            }left++;         
        }
        return ret;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ret;
        queue<TreeNode*> check;
        if (!root)  return ret;
        check.push(root);

        while(!check.empty()){
            int size = check.size();
            vector<int> temp;
            for(int i = 0 ; i < size ; i++){
                TreeNode* top = check.front();
                check.pop();

                temp.push_back(top->val);

                if (top->left) check.push(top->left);
                if (top->right) check.push(top->right);
            }ret.push_back(temp);
        }return ret;
    }
};
